### DISCUSSION TITLE
Set Default ColorPickerType

### DISCUSSION THREAD

**Discussion by imSaharukh** [rydmike/flex_color_picker#60] [Q&A] [answered]
is there a way I can set the default ColorPickerType?

![image](https://user-images.githubusercontent.com/25554766/142598089-05e58114-2acf-4e65-b4b0-629c3c3d5ac2.png)

but the wheel is opening by default 
![image](https://user-images.githubusercontent.com/25554766/142598158-a722d0dd-e92e-4b83-9cca-9bb34ba00d01.png)
but I can't primary to open by default

**Comment by imSaharukh** [comment]
i was using this that's why 

![image](https://user-images.githubusercontent.com/25554766/142605167-b0e6aa6c-fb3e-463b-af15-46c91821c3b0.png)

**Comment by rydmike** [comment, marked as answer]
Hi @imSaharukh, thanks for your question.

The picker will open on the page that has the color you pass to it when you open it. So if you have a box for selecting a color, and you click on it to open the picker, and pass that color it has to the picker when you open it, it will open the picker with the section (picker type) selected where it can first find the color you passed in, including any shade.

So if you passed it a color value that matched e.g. primary Red[200], that will be selected and the Primary picker shown.

If you pass it some custom color that does not exist in Material, Accent or in eg. custom colors you made for the custom tab, then it will always open the wheel and select the color there, that is if you have the wheel enabled.

If wheel is not enabled and there is no matching color on any of the enabled picker types, it will open the first picker type you have enabled.

BR Mike 😃
